main.go

```diff
@@ -126,6 +127,11 @@ func main() {
 
 		result, err := c.Collect()
 		if err != nil {
+			fmt.Printf("Error collecting agent metrics, err: %w\n", err)
+			if errors.Unwrap(err) == collector.ErrUnauthorized {
```

Comment:
This is the conventional and correct way to do this.
```suggestion
			if errors.Is(err, collector.ErrUnauthorized) {
```
There could be many layers of wrapping (there might not be now, but a future refactor could introduce some). So just unwrapping once could miss the error you're looking for.